torch/csrc/jit/script/function_schema_parser.cpp

```diff
@@ -137,6 +137,7 @@ struct SchemaParser {
         name = "";
       }
     } else {
+      L.nextIf('?');
```

Comment:
Adding support for optional sized arrays, e.g. `int[3]?`, will require more changes in the JIT than what are in this patch. In particular tryMatchArgument will need to be modified to correctly handle None values.